subsys/net/ip/tcp.c

```diff
@@ -2042,16 +2050,24 @@ NET_CONN_CB(tcp_established)
 
 	net_context_set_appdata_values(pkt, IPPROTO_TCP);
 
-	data_len = net_pkt_appdatalen(pkt);
-	if (data_len > net_tcp_get_recv_wnd(context->tcp)) {
-		NET_ERR("Context %p: overflow of recv window (%d vs %d), "
-			"pkt dropped",
-			context, net_tcp_get_recv_wnd(context->tcp), data_len);
-		return NET_DROP;
+	if (!context->tcp->zero_wnd_sent) {
```

Comment:
I don't see the purpose of introducing and storing this flag in TCP context. It's not important whether we sent a zero window to the peer or not. The only important thing is what the size of our current receive window when we receive a segment from peer, and that's  net_tcp_get_recv_wnd().